sklearn/decomposition/tests/test_pca.py

```diff
@@ -425,23 +431,25 @@ def test_pca_score3():
 
 
 @pytest.mark.parametrize("svd_solver", PCA_SOLVERS)
-def test_pca_sanity_noise_variance(svd_solver):
+def test_pca_sanity_noise_variance(svd_solver, global_random_seed):
     # Sanity check for the noise_variance_. For more details see
     # https://github.com/scikit-learn/scikit-learn/issues/7568
     # https://github.com/scikit-learn/scikit-learn/issues/8541
     # https://github.com/scikit-learn/scikit-learn/issues/8544
     X, _ = datasets.load_digits(return_X_y=True)
-    pca = PCA(n_components=30, svd_solver=svd_solver, random_state=0)
+    pca = PCA(n_components=30, svd_solver=svd_solver, random_state=global_random_seed)
     pca.fit(X)
     assert np.all((pca.explained_variance_ - pca.noise_variance_) >= 0)
 
 
 @pytest.mark.parametrize("svd_solver", ["arpack", "randomized"])
-def test_pca_score_consistency_solvers(svd_solver):
+def test_pca_score_consistency_solvers(svd_solver, global_random_seed):
     # Check the consistency of score between solvers
     X, _ = datasets.load_digits(return_X_y=True)
-    pca_full = PCA(n_components=30, svd_solver="full", random_state=0)
-    pca_other = PCA(n_components=30, svd_solver=svd_solver, random_state=0)
+    pca_full = PCA(n_components=10, svd_solver="full", random_state=global_random_seed)
```

Comment:
The test was seed-sensitive for the given assertion tolerance when `svd_solver="randomized"` was used. Again, I think this can be expected when comparing the `randomized` solution to the exact solution.
Only comparing the first 10 components (instead of the first 30 components) mitigates this somewhat and the test becomes seed-insensitive in the 0-99 range.